### ISSUE TITLE
Timeout option to Spell.connect/2 is ignored

### ISSUE THREAD

**Issue by clayrichardson** [MyMedsAndMe/spell#52]
This value seems to be ignored: https://github.com/MyMedsAndMe/spell/blob/master/lib/spell.ex#L125

**Comment by jtmoulia** [comment]
True! Happy to look at a PR for this, otherwise I'll get to it in the next 2 weeks.

Thanks for the report, @clayrichardson

**Comment by clayrichardson** [comment]
No problem, I was trying to figure out the best way to fix this, but I am unsure :)

**Comment by clayrichardson** [comment]
I also noticed that when a timeout error happens, there is no retry, thus this seems to be ignored as well:
https://github.com/MyMedsAndMe/spell/blob/master/lib/spell.ex#L103

**Comment by jtmoulia** [comment]
:blush: :bug: -- thanks!

**Comment by clayrichardson** [comment]
Also, I'm not sure if I'm using this lib properly, as I put Spell.connect into a def connect/1 function inside a GenServer, and case on Spell.connect. On {:error, timeout} I call connect/1 again, but it seems to spawn multiple PIDs which send messages to my GenServer's handle_info callback. So it might be retrying somewhere after the lib returns the timeout error.
